presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -142,54 +147,69 @@
 
     @Inject
     public GlueHiveMetastore(HdfsEnvironment hdfsEnvironment, GlueHiveMetastoreConfig glueConfig)
-    {
-        this(hdfsEnvironment, glueConfig, createAsyncGlueClient(glueConfig));
-    }
-
-    public GlueHiveMetastore(HdfsEnvironment hdfsEnvironment, GlueHiveMetastoreConfig glueConfig, AWSGlueAsync glueClient)
     {
         this.hdfsEnvironment = requireNonNull(hdfsEnvironment, "hdfsEnvironment is null");
         this.hdfsContext = new HdfsContext(new ConnectorIdentity(DEFAULT_METASTORE_USER, Optional.empty(), Optional.empty()));
-        this.glueClient = requireNonNull(glueClient, "glueClient is null");
+        this.glueClient = requireNonNull(createAsyncGlueClient(glueConfig,
+                hdfsEnvironment.getConfiguration(hdfsContext, new Path("file:///tmp"))), "glueClient is null");
```

Comment:
if you were usinmg `io.prestosql.plugin.hive.HdfsConfigurationInitializer#initializeConfiguration` instead (with `io.prestosql.plugin.hive.util.ConfigurationUtils#getInitialConfiguration`) then you won't need to put dummy path here.

However, it's better to put classname as config property in `HiveS3Config` and not pass it through `hdfs-config.xml`